### ISSUE TITLE
Critical Security Bug: Editable URL Allows Unauthorized Access to Other Users' Courses

### ISSUE THREAD

**Issue by sak23042006** [DeepanshuMishraa/courseSync#5]
I have identified a severe security vulnerability in the application where users can edit the URL endpoint (Course Number) on the Course Details page to access other users' course information. This issue poses a significant privacy and security risk as it allows unauthorized access to sensitive data.

![Screenshot 2024-06-25 124431](https://github.com/DeepanshuMishraa/courseSync/assets/138766433/44097ad5-e5c5-4600-913c-79bf415b9271)

![Screenshot 2024-06-25 124504](https://github.com/DeepanshuMishraa/courseSync/assets/138766433/4fe79403-9578-4269-9333-66c4ca67bc6b)

check the endpoints of the above given images that gives access to the courses of another user.

**Comment by DeepanshuMishraa** [comment]
Should I assign you this?

**Comment by sak23042006** [comment]
Yes, I can handle this task. When would you like me to complete it?

**Comment by DeepanshuMishraa** [comment]
No hurry, just fix it as fast as you can

**Comment by sak23042006** [comment]
Ya sure, I will fix it as fast as I can. Thanks for letting me know!
